diff --git a/jquery_update.module b/jquery_update.module
index 956c9e5..8507a3d 100644
--- a/jquery_update.module
+++ b/jquery_update.module
@@ -89,7 +89,8 @@ function jquery_update_library_alter(&$javascript, $module) {
     // Replace jQuery with the alternative version.
     $admin_version = variable_get('jquery_update_jquery_admin_version', '');
 
-    if (!empty($admin_version) && path_is_admin(current_path())) {
+    global $theme;
+    if (!empty($admin_version) && path_is_admin(current_path()) && variable_get('admin_theme', FALSE) == $theme && !(function_exists('aer_theme_path_requires_jquery_update') && aer_theme_path_requires_jquery_update(current_path()))) {
       if (version_compare($version, $admin_version, '!=')) {
         $version = $admin_version;
       }
