diff --git a/picture.install b/picture.install
index e891d1d..50097b2 100644
--- a/picture.install
+++ b/picture.install
@@ -83,6 +83,22 @@ function picture_uninstall() {
   variable_del('picture_ckeditor_label');
   variable_del('picture_updated_to_file_entity_2');
   variable_del('picture_js_scope');
+
+  // Delete field formatter settings.
+  ctools_include('export');
+  $entity_info = entity_get_info('file');
+  $file_types = file_type_load_all();
+  $view_modes = array_keys($entity_info['view modes']);
+  $view_modes[] = 'default';
+  $formatters = array('file_picture', 'file_field_picture', 'file_picture_sizes_formatter', 'file_field_picture_sizes_formatter');
+  foreach ($file_types as $file_type) {
+    foreach ($view_modes as $view_mode) {
+      foreach ($formatters as $formatter) {
+        ctools_export_crud_delete('file_display', $file_type->type . '__' . $view_mode . '__' . $formatter);
+      }
+    }
+  }
+  file_info_cache_clear();
 }
 
 /**
